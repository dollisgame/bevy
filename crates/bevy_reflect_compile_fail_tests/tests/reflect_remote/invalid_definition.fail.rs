mod structs {
    use bevy_reflect::reflect_remote;

    mod external_crate {
        pub struct TheirStruct {
            pub value: u32,
        }
    }

    #[reflect_remote(external_crate::TheirStruct)]
    struct MyStruct {
        // Reason: Should be `u32`
        pub value: bool,
    }
}

mod tuple_structs {
    use bevy_reflect::reflect_remote;

    mod external_crate {
        pub struct TheirStruct(pub u32);
    }

    #[reflect_remote(external_crate::TheirStruct)]
    struct MyStruct(
        // Reason: Should be `u32`
        pub bool,
    );
}

mod enums {
    use bevy_reflect::reflect_remote;

    mod external_crate {
        pub enum TheirStruct {
            Unit,
            Tuple(u32),
            Struct { value: usize },
        }
    }

    #[reflect_remote(external_crate::TheirStruct, FromReflect)]
    enum MyStruct {
        // Reason: Should be unit variant
        Unit(i32),
        // Reason: Should be `u32`
        Tuple(bool),
        // Reason: Should be `usize`
        Struct { value: String },
    }
}

fn main() {}
