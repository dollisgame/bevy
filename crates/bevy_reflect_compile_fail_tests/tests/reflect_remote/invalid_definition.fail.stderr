error: unexpected token
  --> tests/reflect_remote/invalid_definition.fail.rs:42:49
   |
42 |     #[reflect_remote(external_crate::TheirStruct, FromReflect)]
   |                                                 ^

error[E0308]: `?` operator has incompatible types
  --> tests/reflect_remote/invalid_definition.fail.rs:10:5
   |
10 |     #[reflect_remote(external_crate::TheirStruct)]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u32`, found `bool`
   |
   = note: `?` operator cannot convert from `bool` to `u32`
   = note: this error originates in the attribute macro `reflect_remote` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/reflect_remote/invalid_definition.fail.rs:13:20
   |
10 |     #[reflect_remote(external_crate::TheirStruct)]
   |     ----------------------------------------------
   |     |
   |     expected due to the type of this binding
   |     in this procedural macro expansion
...
13 |         pub value: bool,
   |                    ^^^^ expected `u32`, found `bool`
   |
   = note: this error originates in the attribute macro `reflect_remote` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: `?` operator has incompatible types
  --> tests/reflect_remote/invalid_definition.fail.rs:24:5
   |
24 |     #[reflect_remote(external_crate::TheirStruct)]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u32`, found `bool`
   |
   = note: `?` operator cannot convert from `bool` to `u32`
   = note: this error originates in the attribute macro `reflect_remote` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/reflect_remote/invalid_definition.fail.rs:27:13
   |
24 |     #[reflect_remote(external_crate::TheirStruct)]
   |     ----------------------------------------------
   |     |
   |     expected due to the type of this binding
   |     in this procedural macro expansion
...
27 |         pub bool,
   |             ^^^^ expected `u32`, found `bool`
   |
   = note: this error originates in the attribute macro `reflect_remote` (in Nightly builds, run with -Z macro-backtrace for more info)
